### ISSUE TITLE
Add exclude argument to Glob()

### ISSUE THREAD

**Issue by bdbaddog** [SCons/scons#2545]
This issue was originally created at: 2010-01-05 06:25:40.
This issue was reported by: `techtonik`.
techtonik said at 2010-01-05 06:25:40
>It is not uncommon that Glob mask must have some files excluded. The most 
>intuitive and simple way would be to add "exclude=None" argument to Glob() that 
>can take string mask, Node or a set of Nodes for exclusion.

gregnoel said at 2010-02-03 14:02:53
>Bug party triage.  Best handled by issue 2199.
> 
>*** This issue has been marked as a duplicate of 2199 ***

techtonik said at 2010-05-12 04:18:41
>intgr on IRC just requested the same feature. Rereading this issue seems like it 
>is not really a duplicate of issue 2199, but depends on it. Changing status 
>accordingly.

intgr said at 2010-05-12 05:34:20
>For some context:
><intgr> Hi all, I'm trying to build a Java JAR/WAR archive for deployment.
>"env.Jar('direc.war', source='WebContent')" works fine, but it would also
>include all .svn directories.
><intgr> What's the "right" way to exclude .svn directories? Is there an env.Glob
>that does exclusions?
><intgr> The WebContent directory contains SVN-versioned files as well as built
>.class files.
><intgr> Another possibility is to use env.Copy to create a "clean" copy of
>WebContent, but I don't see how I can tell it to exclude .svn either
><intgr> Even more annoying that I can't use env.Glob('WebContent/**.class')

intgr said at 2010-05-12 05:38:40
>I need to type a stupid comment to add myself as CC?

gregnoel said at 2010-05-19 06:34:42
>Bug party triage: Steven to put in a quick fix; the real solution will be part
>of adding the IAPAT infrastructure.

gregnoel said at 2010-05-19 06:39:14
>Oops, last comment belongs to another issue; return this one to triage.

gregnoel said at 2010-06-05 18:33:44
>Bug party triage.  Part of the group of related issues that include Glob(),
>FindFile(), and the wildcard functions (issue 2199).

garyo said at 2012-09-01 10:04:01
>de-assigning all tickets assigned to Greg Noel (no longer working on SCons)

bdbaddog said at 2015-09-16 07:43:35
>Feature added in 2.3.5

techtonik said this issue depends on #2199 at 2010-05-12 04:18:41.
